lib/client/util/__mocks__/Geolocation.js

```diff
@@ -0,0 +1,37 @@
+var path = require.requireActual('../Geolocation');
+
+var geoMock = jest.genMockFromModule('../Geolocation');
+
+var _coords = null;
+var _enabled = false;
+function __setLocation(lat, lon) {
+  _enabled = true;
+  _coords = { latitude: lat, longitude: lon };
+}
+
+function __enable() {
+  _enabled = true;
+}
+
+function __disable() {
+  _coords = null;
+  _enabled = false;
+}
+
+function getCurrentPosition(successCallback, errorCallback) {
+  if(_enabled) {
+    if(_coords) {
+      successCallback({ coords: _coords });
+    }
```

Comment:
Yes, I'm simulating a delay in a sense.  I needed to test the behavior of the component while a request was inflight.  

The approach you suggested would work, but I am hesitant to add timeouts to the mock.  The problem is that adding timeouts to tests forces the test writer to know that the mock uses timers internally and know to call jest.runAllTimers() when she wants the callback to be run.  Further, this JS code isn't inherently asynchronous (see http://martinfowler.com/articles/asyncJS.html) and I would add that type of behavior into the test with caution.

It might be a good idea to discuss and set conventions for mocks (when needed) so that we can reason about behavior of mocks as the application grows.

I certainly agree that the names could be better.